bokeh/model.py

```diff
@@ -741,7 +727,21 @@ def __exit__(self, type, value, traceback):
 
     def __enter__(self):
         for model in self._to_remove_after:
-            self._doc.add_root(model)
+            try:
+                self._doc.add_root(model)
+            except RuntimeError as e:
+                try:
+                    for r in takewhile(lambda x: x != model, self._to_remove_after):
+                        r.document.remove_root(r)
+                except RuntimeError:
+                    pass
```

Comment:
Right, I think this might be simpler and also work fine:
```
for r in self._to_remove_after:
    try:
        r.document.remove_root(p)
    except RuntimeError:
        pass
```